Added Linux app store data metadata

Mailaender opened the pull request:
References:
- http://people.freedesktop.org/~hughsient/appdata/
- http://www.freedesktop.org/wiki/Distributions/AppStream/
- http://www.freedesktop.org/software/appstream/docs/index.html

and https://github.com/hedgewars/hw/blob/master/share/hedgewars.appdata.xml

Mailaender committed:
install Linux app store data metadata

pchote commented on packaging/linux/openra.appdata.xml:
```diff
@@ -0,0 +1,46 @@
+  <screenshots>
+    <screenshot type="default">
+      <caption>Red Alert mod</caption>
+      <image height="300" width="500">http://www.openra.net/images/screenshots/ra-base.png</image>
```
You are already linking against a local image for the icon, so why not use the local mod preview images?

Mailaender replied:
That would violate the specification http://www.freedesktop.org/software/appstream/docs/chap-DistroData.html#sect-AppStream-ASXML and use-case (how does the software look like before I install it).

Mailaender replied:
Added a cached and remote icon, too.

pchote replied:
> (how does the software look like before I install it).

I assumed that it would do that the same way it was getting the icon, which I assume from your most recent change is "not" :)

Mailaender replied:
Well, there are other sources such as https://screenshots.debian.net/ but as we will never make it there the regular way offered by AppStream is our only choice.

pchote commented on packaging/linux/openra.appdata.xml:
```diff
@@ -0,0 +1,48 @@
+    </screenshot>
+    <screenshot>
+      <caption>Tiberian Dawn mod</caption>
+      <image height="300" width="600">http://www.openra.net/images/screenshots/20140405-beacons.png</image>
```
I really don't like the idea of linking against a newspost screenshot.  These should be longterm stable urls.  Like I mentioned above, we should use the mod preview images, but uploaded to a stable url such as http://www.openra.net/images/appdata-mod-${MOD}.png

Mailaender replied:
That seems to be a reasonable request. Updated.

Mailaender commented:
This now depends on https://github.com/OpenRA/OpenRAWeb/pull/209.

pchote commented:
The metadata isn't used by the Ubuntu 14.04 software center (before or after installation) or dash.  Are you sure it is set up properly?

Mailaender commented:
Yes, it may need proper repositories on Ubuntu though.

pchote commented:
I find it a bit suspicious that a feature "for distro-agnostic software-centers and software metadata handling" that is backed by Canonical and others doesn't work on Canonical's distro, so I can't give this my +1.

Mailaender commented:
You are probably not installing it right or forgot to update the database. I don't know enough about Ubuntu and also couldn't find any documentation on how to help you with your issue.

pchote commented:
I just ran the packaging script on the head of this PR, and then installed the generated package.  If additional packaging changes are needed, then we will need to do them before merging this.

Mailaender commented:
Be my guest and file a pull request then. I have no experience and don't run Debian/Ubuntu. I also can't guarantee that this will work with Canonical's new invention http://developer.ubuntu.com/en/snappy/

pchote commented:
Contrary to earlier suggestions, I cannot find anywhere in ubuntu 14.04 or 15.04 client that uses this metadata.  There are appdata.xml files for a few gnome applications in `/usr/share/appdata` but the vast majority of application metadata is not included there, so I guess that is still rolled into the distro-wide package metadata package/database.

Unless somebody can prove me wrong, there's no chance of making this work under Ubuntu, so that shouldn't block this.  It also means that I can't test this, however.

pchote commented:
Could you post a screenshot of this in action in OpenSUSE or another distro that supports client-side appdata?

Mailaender commented:
Using https://wiki.gnome.org/Apps/Software I get

before (using an auto-generate appdata.xml from OBS):

![before-gnome-software](https://cloud.githubusercontent.com/assets/756669/7554406/903bf5ba-f728-11e4-8e9b-56e42f71445c.png)

and after (using mine):

![after](https://cloud.githubusercontent.com/assets/756669/7554407/958c84c6-f728-11e4-80e1-d0e0d070f942.png)

The duplicated screenshot appears to be a Zypper AppStream integration bug as it happens with every package.

https://software.opensuse.org/package/openra will also make use of at least one screenshot. However I won't publish my https://build.opensuse.org/package/show/games:openra/development package yet otherwise everyone on that repository will be asked to update.

pchote commented:
Looks good to me, so :+1: based on those screenshots.  We can switch to larger screenshots at some point in the future by modifying the website.

Mailaender commented:
Oh wait, woops I added a copy & paste mistake to the screenshot definitions. Fixed.

You can also clearly see that this was designed by engineers. They even have specifications for screenshot sizes, see bottom of http://people.freedesktop.org/~hughsient/appdata/

obrakmann commented:
:+1: based on that screenshot. Unfortunately Ubuntu's software center doesn't do anything with it, even though the file gets installed in the right place. Maybe 14.04 is too old for that.

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/dcb381b3fc466ac5716179be8b323c14544a24ca%5E...dcb381b3fc466ac5716179be8b323c14544a24ca)

Mailaender commented:
Does it work if you install the GNOME software center on Ubuntu?

obrakmann commented:
It isn't packaged on Ubuntu, [not even on Ubuntu Gnome](http://ubuntugnome.org/documentation/faq/).

Mailaender commented:
Try running `xdg-desktop-menu forceupdate`
